iothub/service/src/Common/Exceptions/DeviceInvalidResultCountException.cs

```diff
@@ -1,37 +1,57 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System;
+using System.Runtime.Serialization;
+
 namespace Microsoft.Azure.Devices.Common.Exceptions
 {
-    using System;
-    using System.Runtime.Serialization;
-
+    /// <summary>
+    /// The exception that is thrown when the count of device results exceeds the specified maximum value.
+    /// Note: This exception is currently not thrown by the client library.
```

Comment:
Not sure what the intention/ use-case is behind this exception. It is currently unreferenced in our code.

Reply:
I'm pretty sure I removed an exception type before that was unreferenced, but that was before the "breaking changes" rule. You could check with Mark about removing it, though. We've removed something else unreferenced since then too.

Reply:
As I clean up the rest of the exception types, I'll keep on adding the unreferenced ones to a list and we can discuss their future